File: wallendar-app/app/src/main/java/com/td/wallendar/groupmembers/GroupMembersAdapter.java
package com.td.wallendar.groupmembers;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.td.wallendar.R;
import com.td.wallendar.models.ApplicationUser;

import java.util.ArrayList;
import java.util.List;

public class GroupMembersAdapter extends RecyclerView.Adapter<GroupMembersViewHolder> {

    private final List<ApplicationUser> dataset;

    public GroupMembersAdapter() {
        this.dataset = new ArrayList<>();
    }

    @NonNull
    @Override
    public GroupMembersViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_group_member, parent, false);
        return new GroupMembersViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull GroupMembersViewHolder holder, int position) {
        holder.bind(dataset.get(position));
    }

    @Override
    public int getItemCount() {
        return dataset == null ? 0 : dataset.size();
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setDataset(final List<ApplicationUser> newDataset) {
        this.dataset.clear();
        if (newDataset != null) {
            this.dataset.addAll(newDataset);
        }
        notifyDataSetChanged();
    }

}

File: wallendar-app/app/src/main/java/com/td/wallendar/groupmembers/GroupMembersAdapter.kt
package com.td.wallendar.groupmembers

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.td.wallendar.R
import com.td.wallendar.models.ApplicationUser
import java.util.*

class GroupMembersAdapter : RecyclerView.Adapter<GroupMembersViewHolder?>() {
    private val dataset: MutableList<ApplicationUser>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GroupMembersViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_group_member, parent, false)
        return GroupMembersViewHolder(view)
    }

    override fun onBindViewHolder(holder: GroupMembersViewHolder, position: Int) {
        holder.bind(dataset[position])
    }

    override fun getItemCount(): Int {
        return dataset.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setDataset(newDataset: MutableList<ApplicationUser>) {
        dataset.clear()
        dataset.addAll(newDataset)
        notifyDataSetChanged()
    }

    init {
        dataset = ArrayList()
    }
}